### ISSUE TITLE
No detection when "smart detection" enabled on a G4 Dome

### ISSUE THREAD

**Issue by Nivlac7** [hjdhjd/homebridge-unifi-protect#671]
**Describe The Problem:**

Thanks a lot for your work, have been using this plugin for a year without any issue! (with G3 only cameras)
Now the issue: Homekit doesn't receive detection as soon as "smart detection" is enabled (on G4 Dome)

**To Reproduce:**
disable the "smart detection" -> detection are received in homekit
enable the "smart detection" -> detection are NOT received in homekit

**Logs:**
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO]: Connected to the UniFi Protect controller API (address: 192.168.2.1 mac: E063DA8CB412).
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO]: Connected to the UniFi Protect realtime update events API.
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO]: Discovered camera: Garage [UVC G3 Flex] (address: 192.168.2.177 mac: 18E829CF9FFB).
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO]: Discovered camera: Buanderie [UVC G3 Flex] (address: 192.168.2.201 mac: 18E829CF9F1B).
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO]: Discovered camera: Jardin [UVC G3 Flex] (address: 192.168.2.247 mac: B4FBE49EF2B1).
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO]: Discovered camera: Entrée Garage [UVC G4 Dome] (address: 192.168.2.79 mac: 68D79AE30E8E).
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO]: Controller refresh interval set to 10 seconds.
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO] Garage [UVC G3 Flex]: HomeKit Secure Video support enabled.
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO] Buanderie [UVC G3 Flex]: HomeKit Secure Video support enabled.
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO] Jardin [UVC G3 Flex]: HomeKit Secure Video support enabled.
[16/02/2022, 16:10:13] [UniFi Protect] UDM Pro Calvin [UDM-PRO] Entrée Garage [UVC G4 Dome]: HomeKit Secure Video support enabled.

[16/02/2022, 16:10:28] [UniFi Protect] UDM Pro Calvin [UDM-PRO] Entrée Garage [UVC G4 Dome]: Streaming request from 192.168.2.133: 1920x1080@30fps, 802 kbps. Using RTSP stream profile: 2688x1512@24fps (High), 10000 kbps.
[16/02/2022, 16:10:51] [UniFi Protect] UDM Pro Calvin [UDM-PRO] Entrée Garage [UVC G4 Dome]: Stopped video streaming session.

```
Show the Homebridge logs here.
Remove any sensitive information.
```

**Homebridge Configuration:**

```json
{
    "controllers": [
        {
            "address": "192.168.2.1",
            "password": "Removed",
            "username": "Removed"
        }
    ],
    "platform": "UniFi Protect"
}
```

**Screenshots:**

**Environment:**

* **Homebridge Version: homebridge -V1.4.0
* **Node Version: v16.14.0
* **Homebridge-UniFi-Protect Plugin Version: 5.3.4
* **UniFi Protect Version: 1.21.0
* **UniFi Protect Controller Type: UDMP
* **UniFi Protect Controller Firmware Version: 1.11.0
* **Apple Device and iOS / macOS / iPadOS / tvOS Version: iPadOS 15.3.1
* **Operating System and OS Version: Raspbian GNU/Linux Buster (10)

**Comment by hjdhjd** [comment]
It's the second two options in your screenshot - Motion Zones and Smart Detection Zones that I have a question about. Check to make sure they're there and they're on? By default, Protect defines a whole-camera zone for each type (motion and smart detection) and sets them to 50% sensitivity.

The issue I think you're alluding it is that there's overlapping capability coverage here:

If you have smart motion detection enabled in HBUP through `Motion.SmartDetect`, HBUP will only alert you to motion events that get triggered through the UniFi Protect controller smart motion capabilities.

HKSV uses that motion indicator to decide when to record a clip and when to alert the end user. If you have `Motion.SmartDetect` enabled **and** you have HKSV recording enabled, you are effectively using two different smart motion detection solutions on top of each other, and you may get strange or undesirable results. In order for a motion alert to get to you in that scenario, both the Protect controller **and** HKSV have to believe there's an object of interest to inform you about.  Hopefully that gives you a bit of a sense of how all this works.

If you have smart motion detection enabled in Protect, and regular motion detection disabled in Protect, and you haven't told HBUP to watch for smart motion events, you will not see motion events.

In short: don't overthink this. Reset to factory defaults on the G4 and configure HKSV and you should be done. 😄 Don't mess with smart motion detection on the Protect controller unless you really need that capability for something else.

So let me ask this: what is it that you're trying to do that you're not able to?

**Comment by hjdhjd** [comment]
Take your time and walk me through it here or on Discord. Happy to help.

**Comment by hjdhjd** [comment]
The documentation is quite clear about what the defaults are. Please read the **extensive** documentation when looking for what capabilities are and are not set by default.

**Comment by Nivlac7** [comment]
Trying to share my humble experience, but I clearly understand your view.

Thanks for the work and assistance :)

**Comment by hjdhjd** [comment]
Always appreciate other perspectives, but equally would appreciate folks reading the documentation and understanding what my perspective is and why I've made the design choices I have. 

I believe software should be opinionated, as I've written in the documentation. You're free to have other opinions...but that doesn't make it a bug or a bad choice for a default. 😄

Glad you're enjoying the work! Lots more to come!
